### PR TITLE
GDB-10950 Use GDB defaults when creating

### PR DESCRIPTION
**Pull request by yordanalexandrov**
## What
GDB API provides default values for agents which should be populated in the create/edit modals

## Why
This will move the responsibility for the default values to GDB where they matter

## How
- added a call to the `defaults` endpoint
- added a prop in the `TTYGContextService` to cache and hold the defaults value
- added a getter in `TTYGContextService` to get the cached default agent or to fetch it from GDB
- refactored the `agentFormModelMapper` to set default values where needed.
- removed hardcoded `AGENT_MODEL_DEFAULT_VALUES` along with the `newAgentFormModelProvider`
- moved default form model values inside the model and set them on initialization
- refactored `onCreateAgent`, `onEditAgent` to first fetch the default values
- added validation errors to `agentSettingsForm` in `agent-settings-modal.controller`, when there are no indexes or connectors

### CODE REVIEW

File: src/js/angular/ttyg/services/ttyg-context.service.js [hunk 1]
```diff
@@ -36,6 +36,27 @@ function TTYGContextService(EventEmitterService) {
...
+    let _defaultAgent = undefined;
+
+    /**
+     * @return {Promise<AgentModel>}
+     */
+    const getDefaultAgent = () => {
+        if (_defaultAgent) {
+            return Promise.resolve(cloneDeep(_defaultAgent));
+        }
+        return TTYGService.getDefaultAgent()
```
Comment (svilenvelikov) [inline src/js/angular/ttyg/services/ttyg-context.service.js:53]: I prefer this to be somewhere else, but not here. Perhaps the ttyg-view.controller is better place. The context service should not deal with loading data

File: src/js/angular/ttyg/controllers/ttyg-view.controller.js [hunk 2]
```diff
@@ -272,38 +279,44 @@ function TTYGViewCtrl(
...
-            (data) => {
-                TTYGService.editAgent(data)
-                    .then((updatedAgent) => {
-                        toastr.success($translate.instant("ttyg.agent.messages.agent_save_successfully", {agentName: updatedAgent.name}));
-                        const hasSelectedAgent = TTYGContextService.getSelectedAgent();
-                        if (hasSelectedAgent && data.id === hasSelectedAgent.id) {
-                            TTYGContextService.selectAgent(updatedAgent);
+        TTYGContextService.getDefaultAgent()
+            .then((agentDefaultValues) => {
+                const agentFormModel = agentFormModelMapper(agentToEdit, agentDefaultValues);
```
Comment (svilenvelikov) [inline src/js/angular/ttyg/controllers/ttyg-view.controller.js:284]: I think it'd be better to call the ttyg.service directly and move the transformation there

File: src/js/angular/ttyg/services/ttyg-context.service.js (hunk 1 above)
Comment (yordanalexandrov) [reply src/js/angular/ttyg/services/ttyg-context.service.js]: Moved it to the `ttyg-view.controller`

File: src/js/angular/ttyg/services/ttyg-context.service.js [hunk 3]
```diff
@@ -47,14 +47,11 @@ function TTYGContextService(EventEmitterService, TTYGService) {
...
     const getDefaultAgent = () => {
-        if (_defaultAgent) {
-            return Promise.resolve(cloneDeep(_defaultAgent));
-        }
-        return TTYGService.getDefaultAgent()
-            .then((response) => {
-                _defaultAgent = response;
-                return cloneDeep(_defaultAgent);
-            });
+        return _defaultAgent;
```
Comment (svilenvelikov) [inline src/js/angular/ttyg/services/ttyg-context.service.js:50]: You forgot to clone it.

**Review by svilenvelikov — APPROVED** [review]

**Comment by yordanalexandrov** [comment]
Jenkins retry build

**Review by svilenvelikov — DISMISSED** [review]

**Comment by svilenvelikov** [comment]
Jenkins retry build

**Comment by svilenvelikov** [comment]
Jenkins retry build

**Comment by svilenvelikov** [comment]
Jenkins retry build

**Review by svilenvelikov — APPROVED** [review]

**Pull request merged**